api/go.mod

```diff
@@ -1,8 +1,8 @@
 module github.com/ramendr/ramen/api
 
-go 1.22.5
+go 1.23.0
```

Comment:
There was nothing about go version in the commit/pr mesages.

Reply:
I updated the Go version here as well, as I thought it would be best to use the same version across all go.mod files in the repository.